homeassistant/components/google_photos/media_source.py

```diff
@@ -158,14 +156,15 @@ async def async_resolve_media(self, item: MediaSourceItem) -> PlayMedia:
         entry = self._async_config_entry(identifier.config_entry_id)
         client = entry.runtime_data
         media_item = await client.get_media_item(media_item_id=identifier.media_id)
-        is_video = media_item["mediaMetadata"].get("video") is not None
+        if media_item.media_metadata and (media_item.media_metadata.video is not None):
+            url = _video_url(media_item)
+            mime_type = media_item.mime_type or "video/mp4"
```

Comment:
How common is to not have mime type? Maybe first try to guess it from the filename before defaulting to mp4 and jpeg below? It might be worth logging when using such a default in case it causes unexpected behavior downstream, e.g. trying to play back quicktime as mp4. Ideally you would have added this default in a separate PR since it isn't relevant to this. If you keep it, mention it in the description.

Reply:
Yeah this is sketchy. I think it's nearly almost always present, but for typing I need to handle a `None` value given this supports field masks and using a generic dataclass. I've updated this to just fail hard if its missing a mime type which seems simplest. (Though i'm not adding test coverage for it)